### DISCUSSION TITLE
Model is downloaded to C: drive which has very little free space,change model download dir

### DISCUSSION THREAD

**Discussion by Theo1996** [openai/whisper#337] [Q&A] [answered]
I installed whisper win 7, when I run it it starts downloading a model (~500mb) to my `C: drive at C:\Users\<user>\.cache\whisper` . That partition is exclusivly for my OS and doent have much free space can I change the dir it downloads to? Do I have to  edit the source code? I dont know python... It would be good if it let you choose where to download files.

**Comment by jianfch** [comment, marked as answer]
you can specify the directory you want to saved the download with `download_root`
```python
import whisper
model = whisper.load_model('base', download_root='some_directory')
```

**Reply by Theo1996** [reply]
Will try it.

**Reply by Theo1996** [reply]
```
    from . import load_model
    model = load_model(model_name, device=device, download_root=model_dir)
```
I found this in transcribe.py ,do I have to just replace model_dir, which file am I supposed to edit?
I think init.py has some relevant lines of code ,where it fownloads the models etc

**Reply by jianfch** [reply]
if you're running it from python you can use what I wrote:
```python
model = whisper.load_model('base', download_root='some_directory')
```
if you're using it from cli change this [line](https://github.com/openai/whisper/blob/d18e9ea5dd2ca57c697e8e55f9e654f06ede25d0/whisper/__init__.py#L93) to:
```python
'''
previously was this:
download_root = os.getenv(
    "XDG_CACHE_HOME", 
    os.path.join(os.path.expanduser("~"), ".cache", "whisper")
)
'''
download_root = 'your_dir_of_choice/'
```
or just use `--model_dir`

**Reply by Theo1996** [reply]
@jianfch --model_dirdoesnt work 
`whisper: error: unrecognized arguments: --model_dir `when I try to use whisper and `no such option: --model_dir` when I try to install it.

**Reply by jianfch** [reply]
`--model_dir` was added in 0b1ba3d46ebf7fe6f953acfd8cad62a4f851b49f. If you don't have those lines, just update whisper.

**Reply by Theo1996** [reply]
@jianfch Thank you very much.

**Comment by Theo1996** [comment]
```
    from . import load_model
    model = load_model(model_name, device=device, download_root=model_dir)
```
I found this in transcribe.py ,do I have to just replace model_dir, which file am I supposed to edit?
I think init.py has some relevant lines of code ,where it fownloads the models etc
